graphql/subcourse/mutations.ts

```diff
@@ -287,6 +288,37 @@ export class MutateSubcourseResolver {
         return true;
     }
 
+    // Join the course from list of prospects; i.e. pupil was a prospect and is now joining the course as requested by the instructor
+    @Mutation((returns) => Boolean)
+    @AuthorizedDeferred(Role.OWNER)
+    async subcourseJoinFromProspects(
+        @Ctx() context: GraphQLContext,
+        @Arg('subcourseId') subcourseId: number,
+        @Arg('pupilId', { nullable: false }) pupilId: number
+    ) {
+        let subcourse = await getSubcourse(subcourseId);
+        await hasAccess(context, 'Subcourse', subcourse);
+        const pupil = await getPupil(pupilId);
+
+        const isProspect = getSubcourseProspects(subcourse).some((p) => p.pupilId === pupilId);
+        if (!isProspect) {
+            throw new PrerequisiteError(`Pupil(${pupil.id}) is not a prospect of the Subcourse(${subcourse.id}) and can thus not be joined by the instructor`);
+        }
+
+        const participantCount = await prisma.subcourse_participants_pupil.count({ where: { subcourseId: subcourse.id } });
```

Comment:
Can we move this into joinSubcourse?

Reply:
We could, but I built it analogous to `subcourseJoinFromWaitinglist`, so if we do that we should also move that into `joinSubcourse` imo

Reply:
mm true, generally would prefer to have all logic in common, Graphql just being the API layer